### PR TITLE
Update to latest api spec and more

### PR DESCRIPTION
**Pull request by stevenvegt**
* Fix integration test
* Add integration test for duplicate key
* Add more errors for duplicate key
* Fix not found error
* Update the api spec and regenerated the client
* Change loglevel for health check output

TODO:
* [x] Remove 400 status on GET and DELETE
* [x] Don't accept the storage of empty secrets
* [ ] Fix the error response from ListKeys when there are no keys for a specific path in Vault (should be an empty list instead).
* [x] Update spec from nuts-node repo where the safeKey has been removed.

### CODE REVIEW

File: api/v1/types.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package v1
+
+import "net/url"
+
+// safeKey is used as the Key type in the API.
+// It implements the TextMarshaler interface to ensure that the key is URL encoded.
```
Comment (woutslakhorst) [inline api/v1/types.go:6]: Should it then also make sure it isn't double escaped?

File: codegen/api-specs/nuts-storage-api-v1.yaml [hunk 2]
```diff
@@ -5,25 +5,66 @@ info:
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ServiceStatus"
+  /secrets:
+    get:
+      operationId: listKeys
+      summary: List all keys in the store
+      description: |
+        List all the keys available in the store.
```
Comment (woutslakhorst) [inline codegen/api-specs/nuts-storage-api-v1.yaml:42]: It lists the Key IDs and not the actual keys? Should be clear from description.

File: codegen/api-specs/nuts-storage-api-v1.yaml [hunk 3]
```diff
@@ -34,15 +75,19 @@ paths:
...
         "404":
-          description: The lookup failed, the secret for the provided path could not be found.
+          description: The lookup failed, the secret for the provided key could not be found.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
         "400":
-          description: The request was invalid. Check the error response for more details.
+          description: The lookup failed. Check the error response for more details.
```
Comment (woutslakhorst) [inline codegen/api-specs/nuts-storage-api-v1.yaml:84]: 400 is a bit weird here. 400s are returned when the client input is wrong, after changing the input the request may be tried again. This request doesn't have client input apart from the path, which always returns a 404 when incorrect?

File: codegen/api-specs/nuts-storage-api-v1.yaml [hunk 4]
```diff
@@ -58,55 +103,96 @@ paths:
...
-          description: The deletion succeeded, the secret for the provided path has been deleted from the key store.
+          description: The deletion succeeded, the secret for the provided key has been deleted from the store.
         "404":
-          description: The deletion failed, the secret for the provided path could not be found.
+          description: The deletion failed, the secret for the provided key could not be found.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
         "400":
```
Comment (woutslakhorst) [inline codegen/api-specs/nuts-storage-api-v1.yaml:130]: Same for this 400

File: api/v1/api.go [hunk 5]
```diff
@@ -68,9 +88,12 @@ func (w Wrapper) StoreSecret(ctx context.Context, request StoreSecretRequestObje
 			Title:   "Could not retrieve stored secret",
 		}), nil
 	}
-	return StoreSecret200JSONResponse{Data: Secret(result)}, nil
+	return StoreSecret200JSONResponse{Secret: Secret(result)}, nil
 }
 
 func (w Wrapper) HealthCheck(ctx context.Context, request HealthCheckRequestObject) (HealthCheckResponseObject, error) {
+	if err := w.vault.Ping(); err != nil {
+		return HealthCheck503JSONResponse{Status: Fail}, nil
```
Comment (woutslakhorst) [inline api/v1/api.go:96]: Do you want to provide additional information on the fail? eg: invalid creds or timeout or host not found?

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

File: codegen/api-specs/nuts-storage-api-v1.yaml (hunk 2 above)
Comment (stevenvegt) [reply codegen/api-specs/nuts-storage-api-v1.yaml:42]: It's a key/value store, were the values are a secret. So a key is always the ID of the secret. Listing the keys does not list the values.

File: .github/workflows/docker-build.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+---
+name: 'Build Docker images'
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*'
```
Comment (reinkrul) [inline .github/workflows/docker-build.yaml:9]: I think the `v` prefix is non-standard for Docker images (we did it wrong for `nuts-node`, I believe). I'd drop it.

File: api/v1/api.go [hunk 7]
```diff
@@ -18,48 +19,84 @@ func NewWrapper(vault vault.Storage) Wrapper {
 func (w Wrapper) DeleteSecret(ctx context.Context, request DeleteSecretRequestObject) (DeleteSecretResponseObject, error) {
 	err := w.vault.DeleteSecret(request.Key)
 	if err != nil {
+		if err == vault.ErrNotFound {
+			return DeleteSecret404JSONResponse{
+				Backend: backend,
+				Detail:  err.Error(),
+				Status:  404,
+				Title:   "could not delete secret",
```
Comment (reinkrul) [inline api/v1/api.go:27]: secret not found?

**Review by reinkrul — APPROVED** [review]
Created https://github.com/nuts-foundation/hashicorp-vault-proxy/issues/8

**Comment by reinkrul** [comment]
Accidentally approved, but had 2 comments

File: api/v1/api.go (hunk 7 above)
Comment (stevenvegt) [reply api/v1/api.go:27]: The error detail contains the "key not found" text. I think the combined information is more clear.
